server/py/services/api/crud/model_monitoring/model_endpoints.py

```diff
@@ -833,76 +835,114 @@ def _add_real_time_metrics(
-        Firstly trying to use project secret and if there is no such secret
-        it's trys to use the default/v3io store connection string.
-        Note : Use this method only for deleting/reading model endpoints.
+        Add basic metrics to the model endpoint object.
+
+        :param model_endpoint_objects: A list of `ModelEndpoint` objects that will
+                                        be filled with the relevant basic metrics.
+        :param project:                The name of the project.
+
+        :return: A list of `ModelEndpointMonitoringMetric` objects.
         """
-        try:
-            model_endpoint_store = (
-                services.api.crud.model_monitoring.helpers.get_store_object(
-                    project=project
-                )
-            )
-        except mlrun.errors.MLRunInvalidMMStoreTypeError:
-            # TODO: delete in 1.9.0 - for BC trying to create default/v3io store
-            store_connection_string = (
-                mlrun.mlconf.model_endpoint_monitoring.endpoint_store_connection
-                or mlrun.common.schemas.model_monitoring.V3IO_MODEL_MONITORING_DB
-                if not mlrun.mlconf.is_ce_mode()
-                else None
+
+        def _add_metric(
+            mep: mlrun.common.schemas.ModelEndpoint,
+            df_dictionary: dict[str, pd.DataFrame],
+        ):
+            for metric in df_dictionary.keys():
+                df = df_dictionary.get(metric, pd.DataFrame())
+                if not df.empty:
+                    line = df[df["endpoint_id"] == mep.metadata.uid]
+                    if not line.empty and metric in line:
+                        value = line[metric].item()
+                        if isinstance(value, pd.Timestamp):
+                            value = value.to_pydatetime()
+                        setattr(mep.status, metric, value)
+
+            return mep
+
+        uids = [mep.metadata.uid for mep in model_endpoint_objects]
+        tsdb_connector = mlrun.model_monitoring.get_tsdb_connector(
+            project=project,
+            secret_provider=services.api.crud.secrets.get_project_secret_provider(
+                project=project
+            ),
+        )
+        error_count_df = tsdb_connector.get_error_count(endpoint_ids=uids)
+        last_request_df = tsdb_connector.get_last_request(endpoint_ids=uids)
+        avg_latency_df = tsdb_connector.get_avg_latency(endpoint_ids=uids)
+        drift_status_df = tsdb_connector.get_drift_status(endpoint_ids=uids)
+
+        return list(
+            map(
+                lambda mep: _add_metric(
+                    mep=mep,
+                    df_dictionary={
+                        "error_count": error_count_df,
+                        "last_request": last_request_df,
+                        "avg_latency": avg_latency_df,
+                        "result_status": drift_status_df,
+                    },
+                ),
+                model_endpoint_objects,
```

Comment:
This could be more elegant. I'm ok with it for now, but this can be improved for maintainability and performance